File: fixtures/mixed-eval/frontend/src/PatientStore.ts
import { Patient } from "./Patient";

// TS-side PatientStore. Intentional collision with Sources/Swift/PatientStore.swift.
// mixed-001 expects both to surface; neither should dominate by >2 ranks.
export class PatientStore {
  private patients: Patient[] = [];

  update(patient: Patient): void {
    this.patients.push(patient);
  }
}

File: fixtures/mixed-eval/Sources/Swift/PatientStore.swift
import SwiftUI
import Foundation

// Swift-side PatientStore. Intentionally collides with the TS PatientStore
// in frontend/src/PatientStore.ts so mixed-001 can verify cross-language
// ranking doesn't unfairly favor either language.
class PatientStore: ObservableObject {
  @Published var patients: [Patient] = []

  func update(id: UUID) {}
  func update(patient: Patient) {}
}
